internal/pkg/term/selector/local_selector.go

```diff
@@ -219,6 +275,42 @@ func (s *localFileSelector) listDockerfiles() ([]string, error) {
 	return dockerfiles, nil
 }
 
+// listDirsAndFiles returns the list of directories and files within the current
+// working directory and two subdirectory levels below.
+func (s *localFileSelector) listDirsAndFiles(workingDir string) ([]string, error) {
+	names, err := s.getDirAndFileNames(workingDir, aws.Int(0), &[]string{})
+	if err != nil {
+		return nil, err
+	} 
+	return names, nil
+}
+
+func (s *localFileSelector) getDirAndFileNames(dir string, depth *int, existingNames *[]string) (allNames []string, err error) {
+	wdDirsAndFiles, err := s.fs.ReadDir(dir)
+	if err != nil {
+		return nil, fmt.Errorf("read directory: %w", err)
+	}
+	if *depth == 0 {
+		dir = s.workingDirAbs
+	}
+	for _, file := range wdDirsAndFiles {
+		name := file.Name()
+		if strings.HasPrefix(name, ".") || name == "copilot" {
+			continue
+		}
+		relPathName := dir + "/" + name
+		*existingNames = append(*existingNames, relPathName)
+		if *depth < 3 && file.IsDir() {
+			*depth += 1
+			_, err := s.getDirAndFileNames(relPathName, depth, existingNames)
+			if err != nil {
+				return nil, fmt.Errorf("get dir and file names: %w", err)
+			}
+		}
+	}
+	return *existingNames, nil
+}
```

Comment:
```suggestion
func (s *localFileSelector) getDirAndFileNames(dir string, depth int) ([]string, error) {
	wdDirsAndFiles, err := s.fs.ReadDir(dir)
	if err != nil {
		return nil, fmt.Errorf("read directory: %w", err)
	}
	var names []string
	for _, file := range wdDirsAndFiles {
		name := file.Name()
		if strings.HasPrefix(name, ".") || name == "copilot" {
			continue
		}
		relPathName := dir + "/" + name
		names = append(names, relPathName)
		if depth < 3 && file.IsDir() {
			subNames, err := s.getDirAndFileNames(relPathName, depth+1)
			if err != nil {
				return nil, fmt.Errorf("get dir and file names: %w", err)
			}
			names = append(names, subNames...)
		}
	}
	return names, nil
}
```
hopefully this makes sense :) i don't think we need the `if` at L293 as dir _is_ `s.workingDirAbs` from L281, but maybe i'm mistaken!